src/decomp_2d.f90

```diff
@@ -117,6 +117,9 @@ module decomp_2d
    ! Shared real/complex buffers
 #if defined(_GPU)
    real(mytype), target, device, allocatable, dimension(:) :: work1, work2
+#if defined(_NCCL)
+   real(mytype), target, device, allocatable, dimension(:) :: work3, work4
```

Comment:
The work arrays `work3` and `work4` could be moved to the module `decomp_2d_nccl`.

They are used in the module `decomp_2d` only when they are allocated. A deallocate statement is thus missing.

Reply:
Yes i think i forgot to deallocate. I'll try to do it now